dynamic_rest/serializers.py

```diff
@@ -301,6 +317,29 @@ def get_name(cls):
         return cls.Meta.name
 
     @classmethod
+    def get_url(self, pk=None):
+        # if associated with a registered viewset, use its URL
+        url = getattr(self, '_url', None)
```

Comment:
It looks like this is always being set in the router, so I think the canonical URL will never get used? Can we try to get the canonical path first and fall back on this if there isn't one?